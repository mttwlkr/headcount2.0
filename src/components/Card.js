import React from 'react';
import '../styles/Card.css';
import PropTypes from 'prop-types';

const Card = ({ info, handleComparison}) => {
  const { location, stats, selected } = info;
  let className = selected ? 'card selected' : 'card';

  const year = Object.keys(stats).map((year, idx) => {
    let className = stats[year] > 0.5 ? 'over-fifty' : 'under-fifty';
    return (  
      <li 
        className={className} 
        key={idx}>
        <span className='year'>{year}</span>
        <span className='card-data'> : {(stats[year] * 100).toFixed(1)}%</span>
      </li> 
    );
  });

  return (
    <article 
      className={className}
      onClick={() => { handleComparison(location); }
      }>
      <h2>{location}</h2>
      <p className='card-label'>Attendance:</p>
      <ul>
        {year}
      </ul>
      <button>{selected ? 'Remove Comparison' : 'Compare District'}</button>
    </article>
  );
};

Card.propTypes = {
  handleComparison: PropTypes.func,
  info: PropTypes.shape({
    location: PropTypes.string,
    stats: PropTypes.object
  }) 
};

export default Card;