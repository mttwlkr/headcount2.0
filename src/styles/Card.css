.card {
  border: 1px solid black;
  width: 250px;
  height: 345px;
  margin: 2% 0;
  background-color: #edeae0;
  color: #6f6d5d;
}

.card h2 {
  text-align: center;
  font-family: 'Libre Baskerville', serif;
}


.card ul {
  list-style: none;
  margin: auto;
  width: 38%;
}

.selected {
  border: 1px solid #e4c64c;
}


/*rgba(128, 26, 63, .7); */