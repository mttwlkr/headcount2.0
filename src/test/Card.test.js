import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from '../components/Card';

describe('Card', () => {
  it('should match the snapshot', () => {
    const info = {location: 'Denver', stats: {'2006': .85 }};
    const wrapper = shallow(<Card info={info} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the correct props', () => {
    const info = {location: 'Denver', stats: {'2006': .85 }};
    const wrapper = mount(<Card info={info} />);
    expect(wrapper.props().info.location).toEqual('Denver');
    expect(wrapper.props().info.stats['2006']).toEqual(.85);
  });

  it('should have over-fifty class if stats.year is >= .50', () => {
    const info1 = {location: 'Denver', stats: {'2006': .85 }};
    const info2 = {location: 'Aspen', stats: {'2006': .15 }};
    const wrapper1 = mount(<Card info={info1} />);
    const wrapper2 = mount(<Card info={info2} />);
    expect(wrapper1.find('li').props().className).toEqual('over-fifty');
    expect(wrapper2.find('li').props().className).toEqual('under-fifty');  
  });
});