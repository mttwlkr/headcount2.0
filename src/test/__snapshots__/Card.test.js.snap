/* eslint-disable */

// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    info={
        Object {
            "data": Object {
              "2006": 0.85,
            },
            "location": "Denver",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Denver
</h2>,
        <ul>
          <li
                    className="over-fifty"
          >
                    2006
                     : 
                    0.85
          </li>
</ul>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Denver",
        },
        "ref": null,
        "rendered": "Denver",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="over-fifty"
>
              2006
               : 
              0.85
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2006",
                " : ",
                0.85,
              ],
              "className": "over-fifty",
            },
            "ref": null,
            "rendered": Array [
              "2006",
              " : ",
              0.85,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Denver
</h2>,
          <ul>
            <li
                        className="over-fifty"
            >
                        2006
                         : 
                        0.85
            </li>
</ul>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Denver",
          },
          "ref": null,
          "rendered": "Denver",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="over-fifty"
>
                2006
                 : 
                0.85
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2006",
                  " : ",
                  0.85,
                ],
                "className": "over-fifty",
              },
              "ref": null,
              "rendered": Array [
                "2006",
                " : ",
                0.85,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
